src/cc2olx/models.py

```diff
@@ -0,0 +1,625 @@
+
+MANIFEST = "imsmanifest.xml"
+DIFFUSE_SHALLOW_SECTIONS = False
+DIFFUSE_SHALLOW_SUBSECTIONS = True
+
+OLX_DIRECTORIES = [
+    "about",
+    "assets",
+    "chapter",
+    "course",
+    "html",
+    "info",
+    "policies",
+    "problem",
+    "sequential",
+    "static",
+    "vertical",
+]
+
+
+def is_leaf(container):
+    return "identifierref" in container
+
+
+def has_only_leaves(container):
+    return all(is_leaf(child) for child in container.get("children", []))
+
+
+def pprint(level, key, value, count=0):
+    return  # Comment this out to see the structure.
+    text = "{spaces}|--> {key}({count}) {value} {title}".format(
+        spaces="    " * level,
+        key=key,
+        value=value["identifier"],
+        count=count,
+        title=value.get("title", "none"),
+    )
+    print(text)
+
+
+class ResourceFile:
+    def __init__(self, href):
+        self.href = href
+
+    def __repr__(self):
+        return "<ResourceFile href={href} />".format(href=self.href, )
+
+
+class ResourceDependency:
+    def __init__(self, identifierref):
+        self.identifierref = identifierref
+
+    def __repr__(self):
+        return "<ResourceDependency identifierref={identifierref} />".format(
+            identifierref=self.identifierref,
+        )
+
+
+class Cartridge:
+    def __init__(self, cartridge_file, workspace):
+        self.cartridge = zipfile.ZipFile(cartridge_file)
+        self.metadata = None
+        self.resources = None
+        self.resources_by_id = {}
+        self.organizations = None
+        self.normalized = None
+        self.version = "1.1"
+        self.file_path = cartridge_file
+        self.directory = None
+        self.ns = {}
+
+        self.workspace = workspace
+
+    def __repr__(self):
+        filename = os.path.basename(self.file_path)
+        text = '<{_class} version="{version}" file="{filename}" />'.format(
+            _class=type(self).__name__, version=self.version, filename=filename,
+        )
+        return text
+
+    def normalize(self):
+        organizations = self.organizations
+        count_organizations = len(organizations)
+        if count_organizations == 0:
+            # This is allowed by the spec.
+            # But what do we do?
+            # Should we build course content out of resources?
+            # -> Mock a flattened section/subsection hierarchy?
+            # -> Create a custom course tab(s)?
+            # -> Dump links on the about page?
+            organization = None
+        elif count_organizations == 1:
+            # This is allowed by the spec.
+            organization = organizations[0]
+        else:
+            # This is _NOT_ allowed by the spec.
+            # We could fall over, but let's just take the first...
+            organization = organizations[0]
+        if not organization:
+            return
+        identifier = organization.get("identifier", "org_1")
+        # structure = organization.get("structure", "rooted-hierarchy")
```

Comment:
Is this commented out code left here for a purpose?

Reply:
Ah, that's from original codebase. Thanks, removed.